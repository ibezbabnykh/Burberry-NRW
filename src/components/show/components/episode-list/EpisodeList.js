import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import dateFormat from 'dateformat';

import './EpisodeList.scss';
import { EPISODE } from 'routes/routesConfig';

const renderEpisodeSeasonItem = (episode) => (
  <tr key={episode.id} className="episode-season-list__row">
    <td className="episode-season-list__row_number">{episode.number}</td>
    <td className="episode-season-list__row_date">{dateFormat(episode.airdate, 'mmmm d, yyyy')}</td>
    <td>
      <Link to={`${EPISODE}/${episode.id}`} className="episode-season-list__row_name">
        {episode.name}
      </Link>
    </td>
  </tr>
);

const renderEpisodeSeason = ({ season, values }) => (
  <section className="episode-season" key={season}>
    <header className="episode-season_title">Season {season}</header>
    <table className="episode-season-list">
      <thead className="episode-season-list__header">
        <tr className="episode-season-list__row">
          <th>Number</th>
          <th>Date</th>
          <th>Name</th>
        </tr>
      </thead>
      <tbody className="episode-season-list__body">{values.map(renderEpisodeSeasonItem)}</tbody>
    </table>
  </section>
);

const EpisodeList = ({ data }) => {
  return (
      <section className="episode-list">
        {data.map(renderEpisodeSeason)}
      </section>
  );
};

EpisodeList.propTypes = {
  data: PropTypes.arrayOf(
    PropTypes.shape({
      season: PropTypes.string.isRequired,
      values: PropTypes.arrayOf(
        PropTypes.shape({
          name: PropTypes.string.isRequired,
          season: PropTypes.number.isRequired,
          number: PropTypes.number.isRequired,
          id: PropTypes.number.isRequired,
          airdate: PropTypes.string.isRequired
        })
      ).isRequired
    })
  )
};

export default EpisodeList;