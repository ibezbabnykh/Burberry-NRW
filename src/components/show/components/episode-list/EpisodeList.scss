@import 'assets/style/variables.scss';

.episode-list {
  
}

.episode-season {
  margin-bottom: 20px;

  &_title {
    font-size: $font-size-h3;
    font-weight: 700;
    color: $color-brand;
    margin-bottom: 10px;
  }

  &-list {
    width: 100%;
    border-collapse: collapse;
    border-spacing: 0;
    border-bottom: 1px solid $color-table-bg;
    font-size: $font-size-h6;

    &__header {
      .episode-season-list__row {
        background-color: $color-table-bg;
      }
    }

    th,
    td {
      padding: 10px 5px 10px 10px;
    }

    &__row {
      text-align: left;

      &_number {
        padding-left: 10px;
      }

      &_name {
        font-weight: 700;
      }

      &_date {
        width: 130px;
      }
    }

    &__body {
      .episode-season-list__row:nth-child(even) {
        background-color: $color-table-bg;
      }
    }
  }
}