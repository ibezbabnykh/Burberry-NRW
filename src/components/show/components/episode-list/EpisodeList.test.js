import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import EpisodeList from './EpisodeList';

configure({ adapter: new Adapter() });

const mockProps = {
  data: [
    {
      season: '1',
      values: [
        { id: 1, number: 1, season: 1, name: 'episode 1x1', airdate: '2016-04-11' },
        { id: 2, number: 2, season: 1, name: 'episode 1x2', airdate: '2016-04-15' }
      ]
    },
    {
      season: '2',
      values: [
        { id: 3, number: 1, season: 2, name: 'episode 2x1', airdate: '2017-04-11' },
        { id: 4, number: 2, season: 2, name: 'episode 2x2', airdate: '2017-04-15' }
      ]
    }
  ]
};

describe('EpisodeList component', () => {
  it('should render', () => {
    const component = toJson(shallow(<EpisodeList {...mockProps} />));
    expect(component).toMatchSnapshot();
  });
});