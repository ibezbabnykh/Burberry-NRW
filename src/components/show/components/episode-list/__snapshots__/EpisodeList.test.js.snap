// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EpisodeList component should render 1`] = `
<section
  className="episode-list"
>
  <section
    className="episode-season"
    key="1"
  >
    <header
      className="episode-season_title"
    >
      Season 
      1
    </header>
    <table
      className="episode-season-list"
    >
      <thead
        className="episode-season-list__header"
      >
        <tr
          className="episode-season-list__row"
        >
          <th>
            Number
          </th>
          <th>
            Date
          </th>
          <th>
            Name
          </th>
        </tr>
      </thead>
      <tbody
        className="episode-season-list__body"
      >
        <tr
          className="episode-season-list__row"
          key="1"
        >
          <td
            className="episode-season-list__row_number"
          >
            1
          </td>
          <td
            className="episode-season-list__row_date"
          >
            April 11, 2016
          </td>
          <td>
            <Link
              className="episode-season-list__row_name"
              to="/episode/1"
            >
              episode 1x1
            </Link>
          </td>
        </tr>
        <tr
          className="episode-season-list__row"
          key="2"
        >
          <td
            className="episode-season-list__row_number"
          >
            2
          </td>
          <td
            className="episode-season-list__row_date"
          >
            April 15, 2016
          </td>
          <td>
            <Link
              className="episode-season-list__row_name"
              to="/episode/2"
            >
              episode 1x2
            </Link>
          </td>
        </tr>
      </tbody>
    </table>
  </section>
  <section
    className="episode-season"
    key="2"
  >
    <header
      className="episode-season_title"
    >
      Season 
      2
    </header>
    <table
      className="episode-season-list"
    >
      <thead
        className="episode-season-list__header"
      >
        <tr
          className="episode-season-list__row"
        >
          <th>
            Number
          </th>
          <th>
            Date
          </th>
          <th>
            Name
          </th>
        </tr>
      </thead>
      <tbody
        className="episode-season-list__body"
      >
        <tr
          className="episode-season-list__row"
          key="3"
        >
          <td
            className="episode-season-list__row_number"
          >
            1
          </td>
          <td
            className="episode-season-list__row_date"
          >
            April 11, 2017
          </td>
          <td>
            <Link
              className="episode-season-list__row_name"
              to="/episode/3"
            >
              episode 2x1
            </Link>
          </td>
        </tr>
        <tr
          className="episode-season-list__row"
          key="4"
        >
          <td
            className="episode-season-list__row_number"
          >
            2
          </td>
          <td
            className="episode-season-list__row_date"
          >
            April 15, 2017
          </td>
          <td>
            <Link
              className="episode-season-list__row_name"
              to="/episode/4"
            >
              episode 2x2
            </Link>
          </td>
        </tr>
      </tbody>
    </table>
  </section>
</section>
`;
